src/zos-io.cc

```diff
@@ -700,6 +714,62 @@ int __disableautocvt(int fd) {
   return fcntl(fd, F_CONTROL_CVT, &req);
 }
 
+int __ccsid_new_file() {
+  char* encode_file_new = getenv("_ENCODE_FILE_NEW");
+
+  int ccsid = 819;
+
+  if (encode_file_new)
+    if (strcmp(encode_file_new, "IBM-1047") == 0) {
+      ccsid = 1047;
+    } else if (strcmp(encode_file_new, "BINARY") == 0) {
+      // Set the file descriptor to binary mode
+      ccsid = FT_BINARY;
+    }
+
+  return ccsid;
+}
+
+int __tag_new_file_filename(int fd, std::string filename) {
+
+ auto it = existing_files.find(filename);
+ if(it != existing_files.end())
+    {
+        unsigned int ccsid = it->second.first;
+        bool txtflag = it->second.second;
+
+        if (ccsid == FT_UNTAGGED && txtflag)
+        {
+           struct f_cnvrt cvtreq = {SETCVTON, 0, 0x0417};
```

Comment:
What does 0x0417 indicate?

Reply:
It's 1047, but in hex. I will refactor this a bit to move the logic to a seperate function __enableautocvt just like the disable one